### PR TITLE
Cart adding items userstory20 #56

### LINKED ISSUE #56: User Story 20, User adds an item to the cart
```
As a visitor or registered user
When I visit an item's show page from the items catalog
I see a link or button to add this item to my cart
And I click its link or button
I am returned to the item index page
I see a flash message indicating the item has been added to my cart
The navigation bar increments my cart counter
```

DO THESE STORIES FIRST:
- User Story 16, Item Show Page

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #57

child of #82

depends on #62

### PR DESCRIPTION
**Pull request by milevy1**
- [] Check this if the PR has been approved by the team to be a quick patch and the rest of this form will not be filled out

# Test Coverage Percentage (Model/Feature):
- rspec - 100%
- rspec spec/models - 100%

# What functionality does this accomplish?
Added Cart functionality.  I followed the class lesson plan to implement and mimicked the syntax.  Please reference the class lesson for questions first.  If you still have questions, let me know. http://backend.turing.io/module2/lessons/cart_implementation

- Includes ability to add items to Cart and all requirements for User Story 20.  
- Also adds Cart counter to nav bar

# What did you struggle on to complete?

# Implements/Fixes:
* description
closes #56 

# Testing Changes
- [x] No Tests have been changed
- [] Some Tests have been changed
- [] All of the Tests have been changed(Please describe what in the world happened)

# Checklist:

- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code

# Please include an emoji of how you feel about this branch:
👻

### CODE REVIEW

File: spec/features/cart/add_items_to_cart_spec.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    it 'correctly increments for multiple items in my cart' do
+      click_on "Add to Cart"
+      visit item_path(@item)
+      click_on "Add to Cart"
+
+      expect(page).to have_content("You now have 2 copies of #{@item.name} in your cart.")
+    end
+
+    it 'shows the total number of items in the cart' do
```
Comment (milevy1) [inline spec/features/cart/add_items_to_cart_spec.rb]: Add test for adding different items

File: spec/models/cart_spec.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+
+RSpec.describe Cart do
+  subject {       # subject is RSpec similar to before :each
+    Cart.new({
+      '1' => 2,   # two copies of item 1
+      '2' => 3    # three copies of item 2
+    })
+  }
+
```
Comment (milevy1) [inline spec/models/cart_spec.rb]: Add describe 'instance methods'

File: spec/models/cart_spec.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  describe "#total_count" do
+    it 'can calculate the total number of items it holds' do
+      expect(subject.total_count).to eq(5)
+    end
+  end
+
+  describe "#add_item" do
+    it 'adds an item to its contents' do
```
Comment (milevy1) [inline spec/models/cart_spec.rb]: Add a new item to test
